mycroft/gui/extensions.py

```diff
@@ -0,0 +1,243 @@
+            bus: MessageBus instance
+            gui: GUI instance
+        """
+
+        self.name = name
+        self.bus = bus
+        self.gui = gui
+        config = Configuration.get()
+        enclosure_config = config.get("enclosure")
+        self.active_extension = enclosure_config.get("extension", "Generic")
+
+        # ToDo: Add Exclusive Support For "Desktop", "Mobile" Extensions
+        self.supported_extensions = ["SmartSpeaker", "Bigscreen", "Generic"]
+
+        if self.active_extension not in self.supported_extensions:
+            self.active_extension = "Generic"
+
+        LOG.info(
+            f"Extensions Manager: Initializing {self.name} with active extension {self.active_extension}")
+        self.activate_extension(self.active_extension)
+
+    def activate_extension(self, extension_id):
+        LOG.info(f"Extensions Manager: Activating Extension {extension_id}")
+
+        # map extension_id to class
+        if extension_id == "SmartSpeaker":
+            self.extension = SmartSpeakerExtension(self.bus, self.gui)
+        elif extension_id == "Bigscreen":
+            self.extension = BigscreenExtension(self.bus, self.gui)
+        else:
+            self.extension = GenericExtension(self.bus, self.gui)
+
+        LOG.info(f"Extensions Manager: Activated Extension {extension_id}")
+        self.bus.emit(
+            Message("extension.manager.activated", {"id": extension_id}))
+
+
+class SmartSpeakerExtension():
+    """ Smart Speaker Extension: This extension is responsible for managing the Smart Speaker
+    specific GUI behaviours. This extension adds support for Homescreens and Homescreen Mangement.
+
+    Args:
+        name: Name of the extension manager
+        bus: MessageBus instance
+        gui: GUI instance
+    """
+
+    def __init__(self, bus, gui):
+        LOG.info("SmartSpeaker: Initializing")
+
+        self.bus = bus
+        self.gui = gui
+        self.homescreen_manager = HomescreenManager(self.bus, self.gui)
+
+        self.homescreen_thread = threading.Thread(
+            target=self.homescreen_manager.run)
+        self.homescreen_thread.start()
+
+        self.device_paired = is_paired()
+        store_conf = ('skill_conf.json')
+
+        self.skill_conf = JsonStorageXDG(store_conf)
+        if not "selected_backend" in self.skill_conf:
+            self.skill_conf["selected_backend"] = "unknown"
+            self.skill_conf.store()
+        else:
+            self.skill_conf = JsonStorageXDG(store_conf)
+
+        try:
+            self.bus.on("ovos.pairing.process.completed",
+                        self.start_homescreen_process)
+            self.bus.on("ovos.pairing.set.backend", self.set_backend_type)
+            self.bus.on("mycroft.gui.screen.close",
+                        self.handle_remove_namespace)
+            self.bus.on("system.reboot", self.handle_system_reboot)
+            self.bus.on("system.shutdown", self.handle_system_shutdown)
```

Comment:
pleae remove these, this functionality belongs in PHAL, shutdown and reboot has nothing to do with gui

Reply:
the PHAL will listen to those system.xxx bus messages, PHAL does all system actions

the shell should only emit them and PHAL will call the actual commands

Reply:
Done